java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java

```diff
@@ -396,7 +394,7 @@ protected void autoAdjustRange() {
                 }
 
                 if (!Double.isNaN(minRange) && !Double.isNaN(maxRange)) {
-                    throw new IllegalStateException("shouldn't happen");
+                    dataRange = new Range(minRange, maxRange);
```

Comment:
Does this need code to check if minRange and maxRange are within dataRange.get{Lower,Upper}Bound()?

Reply:
I don't think so, because dataRange is range determined by datapoints and minRange and maxRange are set by user via min() and max() or range() functions.